src/spaniel-observer.ts

```diff
@@ -54,16 +52,18 @@ export class SpanielObserver implements SpanielObserverInterface {
   thresholds: SpanielThreshold[];
   recordStore: { [key: string]: SpanielRecord; };
   queuedEntries: SpanielObserverEntry[];
+  private parentSpanielInstance: SpanielInstance;
   private paused: boolean;
   private onWindowClosed: () => void;
   private onTabHidden: () => void;
   private onTabShown: () => void;
-  constructor(callback: (entries: SpanielObserverEntry[]) => void, options: SpanielObserverInit = {}) {
+  constructor(callback: (entries: SpanielObserverEntry[]) => void, options: SpanielObserverInit = {}, parentSpanielInstance: SpanielInstance) {
```

Comment:
`parentSpanielInstance` could be simply `spaniel` or (if we go with it) `session`